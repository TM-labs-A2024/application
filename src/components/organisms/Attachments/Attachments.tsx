import { ArrowBackIcon } from '@chakra-ui/icons'
import { IconButton, Text, Heading, Image } from '@chakra-ui/react'
import { isIOS } from '@utils/index'
import { useRouter } from 'next/navigation'
import React from 'react'

export default function Evolution({
  speciality,
  title,
  data,
  type
}: {
  speciality: string
  title: string
  data: {
    description: string
    attachments: { url: string; alt: string }[]
  }
  type: string
}) {
  // --- Hooks -----------------------------------------------------------------
  const router = useRouter()
  // --- END: Hooks ------------------------------------------------------------

  return (
    <div className={`flex h-screen w-screen flex-col p-8 lg:px-96 ${isIOS() ? 'pt-20' : 'pt-8'}`}>
      <div className="mb-8 flex flex-row items-center justify-start gap-4">
        <IconButton
          size="xl"
          aria-label="back"
          variant="link"
          icon={<ArrowBackIcon />}
          onClick={() => {
            router.push(`/especialidad/${speciality}?type=${type}`)
          }}
        />
        <Text className="font-medium">{title}</Text>
      </div>
      <div className="flex h-full flex-col overflow-scroll">
        <div>
          <Heading as="h2" size="sm" mb={4}>
            Descripci√≥n
          </Heading>
          <Text>{data.description}</Text>
        </div>
        <div className={data?.attachments?.length === 0 ? 'hidden' : ''}>
          <Heading as="h2" size="sm" mt={4} mb={4}>
            Archivos adjuntos
          </Heading>
          {data?.attachments?.map((image, idx) => (
            <Image
              key={`order-tests-image-${idx + 1}`}
              src={image.url}
              alt={image.alt}
              mb={4}
              mt={4}
            />
          ))}
        </div>
      </div>
    </div>
  )
}
