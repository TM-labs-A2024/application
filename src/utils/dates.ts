export const formatDate = (date: string) => {
  const chunks = date.split('-')

  const newDate = chunks[1] + '-' + chunks[2] + '-' + chunks[0]

  return newDate
}
