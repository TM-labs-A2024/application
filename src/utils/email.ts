import emailjs from '@emailjs/browser'

export function sendEmail(data: {
  from_name: string
  to_name: string
  code: string | number
  to_email: string
}) {
  const templateParams = {
    ...data
  }

  const serviceKey = process.env.NEXT_PUBLIC_EMAIL_SERVICE_KEY ?? ''
  const templateKey = process.env.NEXT_PUBLIC_EMAIL_TEMPLATE_KEY ?? ''
  const apiKey = process.env.NEXT_PUBLIC_EMAIL_API_KEY ?? ''

  emailjs.send(serviceKey, templateKey, templateParams, apiKey).catch(() => {
    alert('Algo sali√≥ mal, por favor reintenta el registro.')
  })
}
