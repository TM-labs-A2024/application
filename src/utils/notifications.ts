import { GENERIC_ERROR } from '@constants/index'
import { isMobile } from '@utils/index'
import { Store } from 'react-notifications-component'

export const setupErrorNotification = () => {
  Store.addNotification(
    GENERIC_ERROR('Algo salió mal, por favor inténtelo de nuevo.', isMobile(window))
  )
}
