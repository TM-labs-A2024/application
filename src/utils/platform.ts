import { Capacitor } from '@capacitor/core'

export const isWeb = () => Capacitor.getPlatform() === 'web'

export const isIOS = () => Capacitor.getPlatform() === 'ios'

export const isAndroid = () => Capacitor.getPlatform() === 'android'

export const isMobile = (window: { screen: { availWidth: number } }) =>
  Capacitor.getPlatform() === 'ios' ||
  Capacitor.getPlatform() === 'android' ||
  window?.screen?.availWidth < 425
