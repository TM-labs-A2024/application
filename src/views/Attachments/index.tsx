import Attachments from '@src/components/organisms/Attachments'
import React from 'react'

export default function AttachmentsView({
  context
}: {
  context: {
    goBackRef: string
    title: string
    data: {
      description: string
      attachments: { url: string; alt: string }[]
    }
    isPatient: boolean
    isDoctor: boolean
    isOpen: boolean
    description: string
    onClose: () => void
    onDeleteClick: (type: string) => void
    onSubmit: () => void
  }
}) {
  return <Attachments context={context} />
}
