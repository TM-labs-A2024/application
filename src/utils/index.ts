import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
