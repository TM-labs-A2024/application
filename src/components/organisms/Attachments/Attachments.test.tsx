import { render, screen, fireEvent } from '@test/utils/index'

import '@testing-library/jest-dom'
import Attachments from './Attachments'
import 'next/navigation'

import '@testing-library/jest-dom'

jest.mock('next/navigation')

const context = {
  goBackRef: '#',
  title: 'title',
  data: {
    description: 'description',
    attachments: [
      { url: 'https://cdn.pixabay.com/photo/2020/02/02/16/06/arm-4813365_1280.jpg', alt: 'x-ray' }
    ]
  },
  isPatient: false,
  isDoctor: true,
  isOpen: false,
  description: 'description',
  onClose: () => null,
  onDeleteClick: (type: string) => type,
  onSubmit: () => null
}

describe('Organisms > Attachments test', () => {
  test('If the component is rendered in the doctors flow, it shows the deletion buttons', () => {
    render(<Attachments context={context} />)

    expect(screen.getByTestId('delete-button')).toBeInTheDocument()
    expect(screen.getByTestId('delete-image-button')).toBeInTheDocument()
  })

  test('When clicking on the deletion button, the respective confirmation modal is shown', () => {
    render(<Attachments context={context} />)

    const deletionButton = screen.getByTestId('delete-button')
    fireEvent.click(deletionButton)

    expect(screen.getByText(context.description)).toBeInTheDocument()
  })

  test('When clicking on the attachment deletion button, the respective confirmation modal is shown', () => {
    render(<Attachments context={context} />)

    const deletionButton = screen.getByTestId('delete-image-button')
    fireEvent.click(deletionButton)

    expect(screen.getByText(context.description)).toBeInTheDocument()
  })

  test('The component renders the description and images given', () => {
    render(<Attachments context={context} />)

    expect(screen.getByText(context.data.description)).toBeInTheDocument()
    context.data.attachments.forEach((attachment) => {
      expect(screen.getByAltText(attachment.alt)).toBeInTheDocument()
    })
  })

  test('Matches the snapshot', () => {
    render(<Attachments context={context} />)

    expect(screen.getByTestId('attachments')).toMatchSnapshot()
  })
})
