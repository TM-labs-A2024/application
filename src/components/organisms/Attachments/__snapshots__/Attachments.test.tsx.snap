// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms > Attachments test Matches the snapshot 1`] = `
<div
  class="flex h-screen w-screen flex-col p-8 lg:px-96 pt-8"
  data-testid="attachments"
>
  <div
    class="mb-8 flex flex-row items-center gap-4 justify-between"
  >
    <div
      class="flex flex-row gap-4"
    >
      <button
        aria-label="back"
        class="chakra-button css-84vx7q"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            fill="currentColor"
          />
        </svg>
      </button>
      <p
        class="chakra-text font-medium css-0"
      >
        title
      </p>
    </div>
    <button
      aria-label="delete"
      class="chakra-button css-1deodq7"
      data-testid="delete-button"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g
          fill="currentColor"
        >
          <path
            d="M19.452 7.5H4.547a.5.5 0 00-.5.545l1.287 14.136A2 2 0 007.326 24h9.347a2 2 0 001.992-1.819L19.95 8.045a.5.5 0 00-.129-.382.5.5 0 00-.369-.163zm-9.2 13a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zm5 0a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zM22 4h-4.75a.25.25 0 01-.25-.25V2.5A2.5 2.5 0 0014.5 0h-5A2.5 2.5 0 007 2.5v1.25a.25.25 0 01-.25.25H2a1 1 0 000 2h20a1 1 0 000-2zM9 3.75V2.5a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v1.25a.25.25 0 01-.25.25h-5.5A.25.25 0 019 3.75z"
          />
        </g>
      </svg>
    </button>
  </div>
  <div
    class="flex h-full flex-col overflow-scroll"
  >
    <div>
      <h2
        class="chakra-heading css-1p6q0l9"
      >
        Descripci√≥n
      </h2>
      <p
        class="chakra-text css-0"
      >
        description
      </p>
    </div>
    <div
      class=""
    >
      <h2
        class="chakra-heading css-1rc0ucf"
      >
        Archivos adjuntos
      </h2>
      <div
        class="relative w-full"
      >
        <button
          class="absolute right-2 top-2 z-20 rounded-md bg-white px-2 py-1"
          data-testid="delete-image-button"
        >
          <svg
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g
              fill="currentColor"
            >
              <path
                d="M19.452 7.5H4.547a.5.5 0 00-.5.545l1.287 14.136A2 2 0 007.326 24h9.347a2 2 0 001.992-1.819L19.95 8.045a.5.5 0 00-.129-.382.5.5 0 00-.369-.163zm-9.2 13a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zm5 0a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zM22 4h-4.75a.25.25 0 01-.25-.25V2.5A2.5 2.5 0 0014.5 0h-5A2.5 2.5 0 007 2.5v1.25a.25.25 0 01-.25.25H2a1 1 0 000 2h20a1 1 0 000-2zM9 3.75V2.5a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v1.25a.25.25 0 01-.25.25h-5.5A.25.25 0 019 3.75z"
              />
            </g>
          </svg>
        </button>
        <img
          alt="x-ray"
          class=" css-5tsmix"
          src="https://cdn.pixabay.com/photo/2020/02/02/16/06/arm-4813365_1280.jpg"
        />
      </div>
    </div>
  </div>
</div>
`
